app/scripts/components/common/map/controls/aoi/index.tsx

```diff
@@ -75,5 +67,14 @@ export default function DrawControl(props: DrawControlProps) {
     }
   );
 
+  useEffect(() => {
+    if (drawControl?.map) {
```

Comment:
I am having hard time understanding what this does. What is the expected value of `drawControl.map` and when is the case that drawControl has `map` attribute?

Reply:
Sorry about the confusion and I should have double-checked before pushing. I’ve cleaned up the code here and it should make more sense hopefully.

I was looking in the `useControl` [hook](https://github.com/visgl/react-map-gl/blob/7.1-release/src/components/use-control.ts#L28) from the `react-map-gl` repo and saw that `IControl` (the type for `DrawControl`) has `export interface IControl<MapT extends MapInstance = MapInstance>`. And I misread it assuming that `drawControl` had a direct map prop which isn’t actually the case.. The map instance is only passed into the lifecycle methods (like the `onAdd` for example) and it isn’t exposed directly on the `drawControl` object.